### DISCUSSION TITLE
Cookies get deleted after closing installed PWA.

### DISCUSSION THREAD

**Discussion by tomikaka22** [pocketbase/pocketbase#2972] [Q&A] [answered]
I am making a web app, which requires authentication. PocketBase works really well, but I am facing an error I cannot find the solution to.

When I close the browser, the cookies stay, next time I visit the website I am logged in automatically.
When I install the web app as a PWA on desktop, I am able to close the app and open it again while staying logged in.
When I install the web app as a PWA on Firefox for android, I am able to close the app and open it again while staying logged in.

But, when I install it with a Chromium browser on android, I log in, then after closing the app I have to log in again.

I am using Sveltekit, with the default server hooks that make PocketBase work.

I console logged out the cookies, and after logging in they are set, but after closing then launching the PWA they suddenly disappear.
![illustr](https://github.com/pocketbase/pocketbase/assets/46684016/92845e47-48b0-4de4-a2a6-810d52763552)

**Comment by tomikaka22** [comment]
I made a reproduction here: https://github.com/tomikaka22/cookie-bug
Run PocketBase, open the build folder in Node js and install the site as a PWA on android.

**Reply by ganigeorgiev** [reply]
You can try to experiment with the various cookie attributes to see if that will have any effect (see [SerializeOptions](https://github.com/pocketbase/js-sdk/blob/c1da823dcd84674d4fd38e23bb09d8a0514cfa3e/src/stores/utils/cookie.ts#L78) and more specifically `sameSite:lax|strict|none`, `secure:true|false` and `httpOnly:true|false`).

You could also try to reproduce the issue with non-JS SDK exported cookie because I don't think this is related to PocketBase or the JS SDK.

_Unfortunately I don't have much experience with PWAs and I don't think I'll be able to help you._

**Reply by tomikaka22** [reply]
I did some testing, this is what I found:
When I write cookies in chrome then install the PWA, the written cookies are available in the PWA.

If I keep Chrome open next to the PWA, they share their cookies.
This means if I log into PocketBase on Chrome I also log in in the PWA and vice-versa.

If I close chrome but keep the PWA open, I retain my cookies, and I am logged in in 
both the PWA and Chrome.

**If I close the PWA but keep Chrome open, then I reopen the PWA I still retain my cookies, except for the pb_data cookie. 
This means I am logged out in both the PWA and Chrome.**

**But the main thing I found is that the PWA can't persist cookies on it's own.
If Chrome is not running in the background the cookies get reverted to the state they were in when Chrome was last running.**

**Reply by tomikaka22** [reply, marked as answer]
It was SameSite.
The 'strict' setting to be exact.
Setting it to 'none' or 'lax' makes the cookies persist inside PWAs.

**Reply by ganigeorgiev** [reply]
Good to know. If not already, just make sure to also check https://web.dev/samesite-cookies-explained/#explicitly-state-cookie-usage-with-the-samesite-attribute (or any other article) explaining the different modes since in some situations it has security implication.
